ppr-ui/src/components/tables/common/TableRow.vue

```diff
@@ -677,6 +679,13 @@ export default defineComponent({
       return parentReg?.statusType === MhApiStatusTypes.FROZEN
     }
 
+    // locked state for MHR based on API response (eg. TAXN Unit Note is blocking QS)
+    const hasLockedState = (item: MhRegistrationSummaryIF): boolean => {
+      const parentReg: MhRegistrationSummaryIF =
+        item.mhrNumber && getMhRegTableBaseRegs.value?.find(reg => reg.mhrNumber === item.mhrNumber)
```

Comment:
Not something for you to change since its already used in multiple places, but just commenting on this here, so it might be changed in the future.

This seems extremely inefficent. We can pass down the parentReg as prop to the tablerow, instead of needing to go through a loop for every item.

Reply:
Agree, definitely could use some refactoring.